### ISSUE TITLE
explained_variance_std seems unstable

### ISSUE THREAD

**Issue by hermitgreen** [OpenMOSS/Language-Model-SAEs#102]
Hi, I'm training a gemma based SAE model. 

According to [this](https://github.com/OpenMOSS/Language-Model-SAEs/issues/64#issuecomment-2488093612)
> If explained variance starts from ~0.9 and decreases till it plateaus at~0.7 when k decreases to 50, like shown above. That means you are getting everything right.

But my explained_variance_std and loss
<img width="775" alt="Image" src="https://github.com/user-attachments/assets/e12a2c57-4b13-4e4d-b2d9-adfeb2b3b3ba" />

Could you correct my code?

```python
import os
import sys
import time

CUDA_VISIBLE_DEVICES = sys.argv[2]
os.environ["CUDA_VISIBLE_DEVICES"] = CUDA_VISIBLE_DEVICES
os.environ["TOKENIZERS_PARALLELISM"] = "false"

sys.path.append('../src')

import torch

from lm_saes import (
    ActivationFactoryConfig,
    ActivationFactoryDatasetSource,
    ActivationFactoryTarget,
    DatasetConfig,
    InitializerConfig,
    LanguageModelConfig,
    SAEConfig,
    TrainerConfig,
    TrainSAESettings,
    WandbConfig,
    train_sae,
)

if __name__ == "__main__":
    dataset_name = sys.argv[1]
    print(dataset_name)

    settings = TrainSAESettings(
        sae=SAEConfig(
            hook_point_in="blocks.12.hook_resid_post",
            hook_point_out="blocks.12.hook_resid_post",
            d_model=2304,
            expansion_factor=8,
            act_fn="topk",
            norm_activation="token-wise",
            sparsity_include_decoder_norm=True,
            top_k=50,
            dtype=torch.float32,
            device="cuda",
        ),
        initializer=InitializerConfig(
            init_search=True,
            state="training",
        ),
        trainer=TrainerConfig(
            lp=1,
            initial_k=2304 / 2,
            lr=5e-8,
            lr_scheduler_name="constantwithwarmup",
            total_training_tokens=int(1e8),
            lr_warm_up_steps=500,
            log_frequency=100,
            eval_frequency=1000000,
            n_checkpoints=1,
            check_point_save_mode="linear",
            exp_result_path=f"results-{dataset_name}",
        ),
        wandb=WandbConfig(
            wandb_project="interpret",
            exp_name=dataset_name,
        ),
        activation_factory=ActivationFactoryConfig(
            sources=[
                ActivationFactoryDatasetSource(
                    name="fineweb",
                )
            ],
            target=ActivationFactoryTarget.BATCHED_ACTIVATIONS_1D,
            hook_points=["blocks.12.hook_resid_post"],
            batch_size=8192,
            context_size=1024,
            buffer_size=None,
            ignore_token_ids=[],
        ),
        sae_name=f"gemma-2-2b-L12-{dataset_name}",
        sae_series="gemma-2-2b",
        model=LanguageModelConfig(
            model_name="google/gemma-2-2b",
            device="cuda",
            dtype="torch.float32",
        ),
        model_name="gemma-2-2b",
        datasets={
            "fineweb": DatasetConfig(
                dataset_name_or_path=f"HuggingFaceFW/{dataset_name}",
            )
        },
    )
    train_sae(settings)

```

**Comment by hermitgreen** [comment]
I got that LLM has default d_model and it is 768
https://github.com/OpenMOSS/Language-Model-SAEs/blob/main/src/lm_saes/config.py#L306

so while use self model, I must set 
```python
        model=LanguageModelConfig(
            model_name="google/gemma-2-2b",
            d_model=2304,
            device="cuda",
            dtype="torch.float32",
        ),
```

Suggest that remove the default 768 value so users will not confuse~
